doc/whats_new/v0.24.rst

```diff
@@ -130,6 +130,11 @@ Changelog
   :pr:`17090` by :user:`Lisa Schwetlick <lschwetlick>` and
   :user:`Marija Vlajic Wheeler <marijavlajic>`.
 
+- |Efficiency| Reduce memory footprint in :func:`feature_selection._compute_mi_cd`
+  and :func:`feature_selection._compute_mi_cc` by calling :class:`neighbors.KDTree`
```

Comment:
Since the whats new is user facing, highlighting the public functions that this PR improves is preferred:

```suggestion
- |Efficiency| Reduce memory footprint in :func:`feature_selection.mutual_info_classif`
  and :func:`feature_selection.mutual_info_regression` by calling :class:`neighbors.KDTree`
```